src/email.rs

```diff
@@ -1,14 +1,48 @@
-use crate::oauth::determine_provider;
 use crate::templates::load_templates;
 use crate::Args;
 use base64::{self, Engine};
 use lettre::message::header;
 use lettre::transport::smtp::authentication::{Credentials, Mechanism};
 use lettre::{Message, SmtpTransport, Transport};
 use std::error::Error;
+use std::fmt::Display;
 use std::fs;
 use std::io;
 use std::path::Path;
+use std::str::FromStr;
+
+/// An error which can be returned when parsing a provider from an email address.
+#[derive(Debug, PartialEq, Eq, thiserror::Error)]
+#[error("No supported email provider for '{0}'")]
+pub struct ParseProviderError(String);
```

Comment:
Would love your thoughts on the advantages of representing this error as a struct, versus a variant that is a part of an "EmailError" enum.

Reply:
A larger `EmailError` enum might make sense, but I was trying to solve one problem at a time and tend to start with a `struct` when I have only a single error type.

Your `send_email` function could ultimately return a `EmailError` enum instead of an `io::Error` and that might be a good thing. However, I'm not sure this particular error makes sense to encounter in that function. Right now I think it might belong in the UI logic before you even spawn a new thread to send the email.